src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/ISolutionExtensions.cs

```diff
@@ -56,5 +56,16 @@ public static Document GetRequiredDocument(this Solution solution, DocumentId do
 
             return document;
         }
+
+        public static Document GetRequiredDocument(this Project project, DocumentId documentId)
+        {
+            var document = project.GetDocument(documentId);
+            if (document == null)
```

Comment:
Embrace the `is null` lifestyle :smile:

Reply:
![image](https://user-images.githubusercontent.com/201340/76889629-2a4cdf80-6843-11ea-8d4f-f6178ca9fa36.png)

Reply:
![image](https://user-images.githubusercontent.com/146967/76892689-56b72a80-6848-11ea-845f-7f83bf92f84a.png)

Reply:
Let me fix that for you:

`=> project.GetDocument(documentId) ?? throw new InvalidOperationException(WorkspaceExtensionsResources.The_solution_does_not_contain_the_specified_document)`

---
In reply to: [393909656](https://github.com/dotnet/roslyn/pull/42373#discussion_r393909656) [](ancestors = 393909656)